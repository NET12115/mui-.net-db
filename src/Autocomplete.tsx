import React, { ChangeEvent } from 'react';

import { Field, FieldRenderProps, FieldProps } from 'react-final-form';

import TextField, { TextFieldProps as MuiTextFieldProps } from '@material-ui/core/TextField';
import {
	default as MuiAutocomplete,
	AutocompleteProps as MuiAutocompleteProps,
	RenderInputParams as MuiAutocompleteRenderInputParams,
} from '@material-ui/lab/Autocomplete';

export type AutocompleteData = {
	[key: string]: any;
};

interface AutocompleteProps extends Partial<MuiAutocompleteProps> {
	name: string;
	label: string;
	required?: boolean;
	getOptionValue: (option: any) => any;
	options: AutocompleteData[];
	fieldProps?: FieldProps<any, any>;
	textFieldProps?: Partial<MuiTextFieldProps>;
}

export const Autocomplete = (props: AutocompleteProps) => {
	const { name, fieldProps, ...rest } = props;

	return (
		<Field
			name={name}
			render={fieldRenderProps => <AutocompleteWrapper {...fieldRenderProps} {...rest} />}
			{...fieldProps}
		/>
	);
};

interface MuiRffAutocompleteProps extends FieldRenderProps<MuiTextFieldProps, HTMLElement> {
	label: string;
	required?: boolean;
	multiple?: boolean;
	textFieldProps?: Partial<MuiTextFieldProps>;
	getOptionValue: (option: any) => any;
}

const AutocompleteWrapper = (props: MuiRffAutocompleteProps) => {
	const {
		input: { name, onChange, value, ...restInput },
		meta,
		label,
		required,
		multiple,
		textFieldProps,
		getOptionValue,
		...rest
	} = props;

	const showError = ((meta.submitError && !meta.dirtySinceLastSubmit) || meta.error) && meta.touched;
	const { variant, ...restTextFieldProps } = (textFieldProps as any) || {};

	return (
		<MuiAutocomplete
			multiple={multiple}
			onChange={
				multiple
					? (_e: ChangeEvent<{}>, values: any[]): void => {
							onChange(values.map(getOptionValue));
					  }
					: (_e: ChangeEvent<{}>, option: any): void => {
							option ? onChange(getOptionValue(option)) : onChange(undefined);
					  }
			}
			renderInput={(params: MuiAutocompleteRenderInputParams) => (
				<TextField
					label={label}
					required={required}
					margin="normal"
					fullWidth
					error={showError}
					helperText={showError ? meta.error || meta.submitError : undefined}
					variant={variant}
					{...params}
					{...restInput}
					{...restTextFieldProps}
				/>
			)}
			{...rest}
		/>
	);
};
