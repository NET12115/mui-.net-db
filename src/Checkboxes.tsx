import {
	Checkbox as MuiCheckbox,
	FormControl,
	FormControlLabel,
	FormGroup,
	FormHelperText,
	FormLabel,
} from '@material-ui/core';
import { CheckboxProps } from '@material-ui/core/Checkbox';
import { FormControlProps } from '@material-ui/core/FormControl';
import React from 'react';

import { Field, FieldRenderProps } from 'react-final-form';

export interface CheckboxData {
	label: string;
	value: string;
}

export interface CheckboxesProps {
	required: boolean;
	label: string;
	name: string;
	data: CheckboxData[];
	error: string;
	fieldProps?: FieldRenderProps<CheckboxProps, HTMLInputElement>;
	formControlProps?: FormControlProps;
}

export function Checkboxes(props: CheckboxesProps) {
	const {
		required,
		label,
		data,
		name,
		error,
		fieldProps,
		formControlProps,
	} = props;

	return (
		<FormControl
			required={required}
			error={!!error}
			margin="normal"
			{...formControlProps}
		>
			<FormLabel>{label}</FormLabel>
			<FormGroup>
				{data.map((item: CheckboxData, idx: number) => (
					<FormControlLabel
						key={idx}
						label={item.label}
						value={item.value}
						control={
							<Field
								component={CheckboxWrapper}
								type="checkbox"
								name={name}
								{...fieldProps}
							/>
						}
					/>
				))}
			</FormGroup>
			{error ? <FormHelperText>{error}</FormHelperText> : <></>}
		</FormControl>
	);
}

function CheckboxWrapper(
	props: FieldRenderProps<CheckboxProps, HTMLInputElement>
) {
	const {
		input: { name, checked, onChange, ...restInput },
		meta,
		...rest
	} = props;
	return (
		<MuiCheckbox
			name={name}
			checked={checked}
			onChange={onChange}
			inputProps={restInput as any}
			{...rest}
		/>
	);
}
