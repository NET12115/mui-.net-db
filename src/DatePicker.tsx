import React from 'react';

import 'date-fns';
import DateFnsUtils from '@date-io/date-fns';

import {
	KeyboardDatePicker,
	KeyboardDatePickerProps,
	MuiPickersUtilsProvider,
} from '@material-ui/pickers';

import { Field, FieldRenderProps } from 'react-final-form';

export function DatePicker(props: Partial<KeyboardDatePickerProps>) {
	const { name } = props;

	return (
		<Field
			name={name as any}
			render={({ input, meta }) => (
				<DatePickerWrapper input={input} meta={meta} {...props} />
			)}
		/>
	);
}

function DatePickerWrapper(
	props: FieldRenderProps<KeyboardDatePickerProps, HTMLElement>
) {
	const {
		input: { name, onChange, value, ...restInput },
		meta,
		...rest
	} = props;
	const showError =
		((meta.submitError && !meta.dirtySinceLastSubmit) || meta.error) &&
		meta.touched;

	return (
		<MuiPickersUtilsProvider utils={DateFnsUtils}>
			<KeyboardDatePicker
				disableToolbar
				fullWidth={true}
				autoOk={true}
				helperText={showError ? meta.error || meta.submitError : undefined}
				error={showError}
				variant="inline"
				format="yyyy-MM-dd"
				margin="normal"
				onChange={onChange}
				value={(value as any) === '' ? null : value}
				{...rest}
				inputProps={restInput}
			/>
		</MuiPickersUtilsProvider>
	);
}
