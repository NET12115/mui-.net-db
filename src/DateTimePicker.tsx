import React from 'react';

import { DateTimePicker as MuiDateTimePicker, DateTimePickerProps as MuiDateTimePickerProps } from '@mui/lab';

import { Field, FieldProps, FieldRenderProps } from 'react-final-form';

import { ShowErrorFunc, showErrorOnChange } from './Util';
import { TextField } from '@mui/material';
import pickerProviderWrapper from './PickerProvider';

export interface DateTimePickerProps extends Partial<Omit<MuiDateTimePickerProps, 'onChange'>> {
	name: string;
	dateFunsUtils?: any;
	locale?: any;
	fieldProps?: Partial<FieldProps<any, any>>;
	showError?: ShowErrorFunc;
}

export function DateTimePicker(props: DateTimePickerProps) {
	const { name, fieldProps, ...rest } = props;

	return (
		<Field
			name={name}
			render={(fieldRenderProps) => <DateTimePickerWrapper {...fieldRenderProps} {...rest} />}
			{...fieldProps}
		/>
	);
}

interface DateTimePickerWrapperProps extends FieldRenderProps<MuiDateTimePickerProps> {
	dateFunsUtils?: any;
	locale?: any;
}

function DateTimePickerWrapper(props: DateTimePickerWrapperProps) {
	const {
		input: { name, onChange, value, ...restInput },
		meta,
		dateFunsUtils,
		locale,
		showError = showErrorOnChange,
		...rest
	} = props;

	const { error, submitError } = meta;
	const isError = showError({ meta });

	const { helperText, ...lessrest } = rest;

	return pickerProviderWrapper(
		dateFunsUtils,
		<MuiDateTimePicker
			onChange={onChange}
			value={(value as any) === '' ? null : value}
			{...lessrest}
			renderInput={(props) => (
				<TextField
					fullWidth={true}
					helperText={isError ? error || submitError : helperText}
					error={isError}
					name={name}
					{...restInput}
					{...props}
				/>
			)}
		/>,
		locale,
	);
}
