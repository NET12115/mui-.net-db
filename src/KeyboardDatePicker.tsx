import React from 'react';

import {
	KeyboardDatePicker as MuiKeyboardDatePicker,
	KeyboardDatePickerProps as MuiKeyboardDatePickerProps,
	MuiPickersUtilsProvider,
} from '@material-ui/pickers';

import { Field, FieldRenderProps } from 'react-final-form';

interface KeyboardDatePickerProps extends Partial<MuiKeyboardDatePickerProps> {
	dateFunsUtils: any;
}

export function KeyboardDatePicker(props: KeyboardDatePickerProps) {
	const { name } = props;

	return (
		<Field
			name={name as any}
			render={fieldRenderProps => (
				<KeyboardDatePickerWrapper {...fieldRenderProps} {...props} />
			)}
		/>
	);
}

interface DatePickerWrapperProps
	extends FieldRenderProps<MuiKeyboardDatePickerProps, HTMLElement> {
	dateFunsUtils: any;
}

function KeyboardDatePickerWrapper(props: DatePickerWrapperProps) {
	const {
		input: { name, onChange, value, ...restInput },
		meta,
		dateFunsUtils,
		...rest
	} = props;

	const showError =
		((meta.submitError && !meta.dirtySinceLastSubmit) || meta.error) &&
		meta.touched;

	return (
		<MuiPickersUtilsProvider utils={dateFunsUtils}>
			<MuiKeyboardDatePicker
				disableToolbar
				fullWidth={true}
				autoOk={true}
				helperText={showError ? meta.error || meta.submitError : undefined}
				error={showError}
				variant="inline"
				format="yyyy-MM-dd"
				margin="normal"
				onChange={onChange}
				value={(value as any) === '' ? null : value}
				{...rest}
				inputProps={restInput}
			/>
		</MuiPickersUtilsProvider>
	);
}
