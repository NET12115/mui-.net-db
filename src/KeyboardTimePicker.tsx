import React from 'react';

import {
	KeyboardTimePicker as MuiKeyboardTimePicker,
	KeyboardTimePickerProps as MuiKeyboardTimePickerProps,
} from '@material-ui/pickers';

import { Field, FieldProps, FieldRenderProps } from 'react-final-form';

import pickerProviderWrapper from './PickerProvider';

export interface KeyboardTimePickerProps extends Partial<MuiKeyboardTimePickerProps> {
	dateFunsUtils?: any;
	fieldProps?: FieldProps<any, any>;
}

export function KeyboardTimePicker(props: KeyboardTimePickerProps) {
	const { name, fieldProps, ...rest } = props;

	return (
		<Field
			name={name as any}
			render={fieldRenderProps => <KeyboardTimePickerWrapper {...fieldRenderProps} {...rest} />}
			{...fieldProps}
		/>
	);
}

interface KeyboardTimePickerWrapperProps extends FieldRenderProps<MuiKeyboardTimePickerProps, HTMLElement> {
	dateFunsUtils?: any;
}

function KeyboardTimePickerWrapper(props: KeyboardTimePickerWrapperProps) {
	const {
		input: { name, onChange, value, ...restInput },
		meta,
		dateFunsUtils,
		...rest
	} = props;

	const showError = ((meta.submitError && !meta.dirtySinceLastSubmit) || meta.error) && meta.touched;

	return pickerProviderWrapper(
		dateFunsUtils,
		<MuiKeyboardTimePicker
			fullWidth={true}
			autoOk={true}
			helperText={showError ? meta.error || meta.submitError : undefined}
			error={showError}
			margin="normal"
			onChange={onChange}
			name={name}
			value={(value as any) === '' ? null : value}
			{...rest}
			inputProps={restInput}
		/>
	);
}
