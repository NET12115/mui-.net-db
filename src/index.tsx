export { Autocomplete, AutocompleteData, AutocompleteProps } from './Autocomplete';
export { Checkboxes, CheckboxData, CheckboxesProps } from './Checkboxes';
export { Switches, SwitchData, SwitchesProps as SwitchProps } from './Switches';
export { Radios, RadioData, RadiosProps } from './Radios';
export { Select, SelectData, SelectProps } from './Select';
export { KeyboardDatePicker, KeyboardDatePickerProps } from './KeyboardDatePicker';
export { DatePicker, DatePickerProps } from './DatePicker';
export { KeyboardTimePicker, KeyboardTimePickerProps } from './KeyboardTimePicker';
export { TimePicker, TimePickerProps } from './TimePicker';
export { TextField, TextFieldProps } from './TextField';
export { makeValidate, makeRequired } from './Validation';
export { Debug } from './Debug';
