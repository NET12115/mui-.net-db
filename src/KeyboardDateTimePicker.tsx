import { DateTimePicker, DateTimePickerProps } from './DateTimePicker';

/**
 * @deprecated use DateTimePicker instead
 */
export const KeyboardDateTimePicker = DateTimePicker;

/**
 * @deprecated use DateTimePickerProps instead
 */
export type KeyboardDateTimePickerProps = DateTimePickerProps;
