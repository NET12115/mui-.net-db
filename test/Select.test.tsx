import { MenuItem } from '@material-ui/core';
import {
	createGenerateClassName,
	StylesProvider,
} from '@material-ui/core/styles';
import { render } from '@testing-library/react';
import React from 'react';
import { Form } from 'react-final-form';

import { Select, SelectData } from '../src';

interface ComponentProps {
	data: SelectData[];
	initialValues: FormData;
	validator?: any;
}

interface FormData {
	best: string;
}

describe('Select', () => {
	const selectData: SelectData[] = [
		{ label: 'Ack', value: 'ack' },
		{ label: 'Bar', value: 'bar' },
		{ label: 'Foo', value: 'foo' },
	];

	const initialValues: FormData = {
		best: 'bar',
	};

	function SelectComponent({ initialValues, data, validator }: ComponentProps) {
		// make a copy of the data because the state is mutated below in one of the tests for clicks
		// then the state is used again for comparison later, which causes tests to be dependent on execution
		// order and fail.
		const generateClassName = createGenerateClassName({
			disableGlobal: true,
			productionPrefix: 'test',
		});

		const onSubmit = (values: FormData) => {
			console.log(values);
		};

		const validate = async (values: FormData) => {
			if (validator) {
				return validator(values);
			}
		};

		return (
			<StylesProvider generateClassName={generateClassName}>
				<Form
					onSubmit={onSubmit}
					initialValues={initialValues}
					validate={validate}
					render={({ handleSubmit }) => (
						<form onSubmit={handleSubmit} noValidate>
							<Select label="Test" required={true} name="best" data={data} />
						</form>
					)}
				/>
			</StylesProvider>
		);
	}

	function SelectComponentMenuItem({
		initialValues,
		data,
		validator,
	}: ComponentProps) {
		// make a copy of the data because the state is mutated below in one of the tests for clicks
		// then the state is used again for comparison later, which causes tests to be dependent on execution
		// order and fail.
		const generateClassName = createGenerateClassName({
			disableGlobal: true,
			productionPrefix: 'test',
		});

		const onSubmit = (values: FormData) => {
			console.log(values);
		};

		const validate = async (values: FormData) => {
			if (validator) {
				return validator(values);
			}
		};

		return (
			<StylesProvider generateClassName={generateClassName}>
				<Form
					onSubmit={onSubmit}
					initialValues={initialValues}
					validate={validate}
					render={({ handleSubmit }) => (
						<form onSubmit={handleSubmit} noValidate>
							<Select label="Test" required={true} name="best">
								{data.map(item => (
									<MenuItem value={item.value} key={item.value}>
										{item.label}
									</MenuItem>
								))}
							</Select>
						</form>
					)}
				/>
			</StylesProvider>
		);
	}

	it('renders without errors', () => {
		const rendered = render(
			<SelectComponent data={selectData} initialValues={initialValues} />
		);
		expect(rendered).toMatchSnapshot();
	});

	it('renders using menu items', async () => {
		const rendered = render(
			<SelectComponentMenuItem
				data={selectData}
				initialValues={initialValues}
			/>
		);

		const form = await rendered.findByRole('form');
		const input = form
			.getElementsByTagName('input')
			.item(0) as HTMLInputElement;
		expect(input.value).toBe('bar');

		expect(rendered).toMatchSnapshot();
	});

	it('renders a selected item', async () => {
		const rendered = render(
			<SelectComponent data={selectData} initialValues={initialValues} />
		);

		const form = await rendered.findByRole('form');
		const input = form
			.getElementsByTagName('input')
			.item(0) as HTMLInputElement;
		expect(input.value).toBe('bar');
	});

	it('has the Test label', () => {
		const rendered = render(
			<SelectComponent data={selectData} initialValues={initialValues} />
		);
		const elem = rendered.getByText('Test') as HTMLLegendElement;
		expect(elem.tagName).toBe('LABEL');
	});

	it('has the required *', () => {
		const rendered = render(
			<SelectComponent data={selectData} initialValues={initialValues} />
		);
		const elem = rendered.getByText('*') as HTMLSpanElement;
		expect(elem.tagName).toBe('SPAN');
		expect(elem.innerHTML).toBe('â€‰*');
	});

	it('requires something selected', async () => {
		// const message = 'something for testing';
		//
		// const validateSchema = makeValidate(
		// 	Yup.object().shape({
		// 		best: Yup.string().required(message),
		// 	})
		// );
		//
		// const rendered = render(
		// 	<SelectComponent
		// 		data={selectData}
		// 		validator={validateSchema}
		// 		initialValues={initialValues}
		// 	/>
		// );
		// TODO: write this test...
	});
});
