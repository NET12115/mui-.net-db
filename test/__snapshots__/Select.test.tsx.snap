// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select renders using menu items 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        novalidate=""
      >
        <div
          class="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
        >
          <label
            class="MuiFormLabel-root-17 MuiInputLabel-root-5 MuiInputLabel-formControl-11 MuiInputLabel-animated-14 MuiInputLabel-shrink-13 MuiFormLabel-filled-21 MuiFormLabel-required-22 MuiInputLabel-required-9"
            data-shrink="true"
            for="best"
          >
            Test
            <span
              class="MuiFormLabel-asterisk-23 MuiInputLabel-asterisk-10"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root-45 MuiInput-root-33 MuiInput-underline-37 MuiInputBase-formControl-46 MuiInput-formControl-34"
          >
            <div
              aria-haspopup="listbox"
              class="MuiSelect-root-24 MuiSelect-select-25 MuiSelect-selectMenu-28 MuiInputBase-input-55 MuiInput-input-41 MuiInputBase-inputSelect-59"
              id="select-best"
              role="button"
              tabindex="0"
            >
              Bar
            </div>
            <input
              name="best"
              type="hidden"
              value="bar"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-65 MuiSelect-icon-30"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      novalidate=""
    >
      <div
        class="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
      >
        <label
          class="MuiFormLabel-root-17 MuiInputLabel-root-5 MuiInputLabel-formControl-11 MuiInputLabel-animated-14 MuiInputLabel-shrink-13 MuiFormLabel-filled-21 MuiFormLabel-required-22 MuiInputLabel-required-9"
          data-shrink="true"
          for="best"
        >
          Test
          <span
            class="MuiFormLabel-asterisk-23 MuiInputLabel-asterisk-10"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root-45 MuiInput-root-33 MuiInput-underline-37 MuiInputBase-formControl-46 MuiInput-formControl-34"
        >
          <div
            aria-haspopup="listbox"
            class="MuiSelect-root-24 MuiSelect-select-25 MuiSelect-selectMenu-28 MuiInputBase-input-55 MuiInput-input-41 MuiInputBase-inputSelect-59"
            id="select-best"
            role="button"
            tabindex="0"
          >
            Bar
          </div>
          <input
            name="best"
            type="hidden"
            value="bar"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-65 MuiSelect-icon-30"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Select renders without errors 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        novalidate=""
      >
        <div
          class="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
        >
          <label
            class="MuiFormLabel-root-17 MuiInputLabel-root-5 MuiInputLabel-formControl-11 MuiInputLabel-animated-14 MuiInputLabel-shrink-13 MuiFormLabel-filled-21 MuiFormLabel-required-22 MuiInputLabel-required-9"
            data-shrink="true"
            for="best"
          >
            Test
            <span
              class="MuiFormLabel-asterisk-23 MuiInputLabel-asterisk-10"
            >
               
              *
            </span>
          </label>
          <div
            class="MuiInputBase-root-45 MuiInput-root-33 MuiInput-underline-37 MuiInputBase-formControl-46 MuiInput-formControl-34"
          >
            <div
              aria-haspopup="listbox"
              class="MuiSelect-root-24 MuiSelect-select-25 MuiSelect-selectMenu-28 MuiInputBase-input-55 MuiInput-input-41 MuiInputBase-inputSelect-59"
              id="select-best"
              role="button"
              tabindex="0"
            >
              Bar
            </div>
            <input
              name="best"
              type="hidden"
              value="bar"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root-65 MuiSelect-icon-30"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      novalidate=""
    >
      <div
        class="MuiFormControl-root-1 MuiFormControl-marginNormal-2 MuiFormControl-fullWidth-4"
      >
        <label
          class="MuiFormLabel-root-17 MuiInputLabel-root-5 MuiInputLabel-formControl-11 MuiInputLabel-animated-14 MuiInputLabel-shrink-13 MuiFormLabel-filled-21 MuiFormLabel-required-22 MuiInputLabel-required-9"
          data-shrink="true"
          for="best"
        >
          Test
          <span
            class="MuiFormLabel-asterisk-23 MuiInputLabel-asterisk-10"
          >
             
            *
          </span>
        </label>
        <div
          class="MuiInputBase-root-45 MuiInput-root-33 MuiInput-underline-37 MuiInputBase-formControl-46 MuiInput-formControl-34"
        >
          <div
            aria-haspopup="listbox"
            class="MuiSelect-root-24 MuiSelect-select-25 MuiSelect-selectMenu-28 MuiInputBase-input-55 MuiInput-input-41 MuiInputBase-inputSelect-59"
            id="select-best"
            role="button"
            tabindex="0"
          >
            Bar
          </div>
          <input
            name="best"
            type="hidden"
            value="bar"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root-65 MuiSelect-icon-30"
            focusable="false"
            role="presentation"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
